lambdas/service/app.py

```diff
@@ -762,204 +757,252 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/index/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/index/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+
+    filter_schema = schema.object_type(
+        default="{}",
+        example={'cellCount': {'within': [[10000, 1000000000]]}},
+        properties={
+            facet: {
+                'oneOf': [
+                    schema.object(is_=schema.array({})),
+                    *[
+                        schema.object_type(properties={op: array_pair_spec})
+                        for op in ['contains', 'within', 'intersects']
+                    ]
+                ]
+            }
+            for facet in facets
+        }
+    )
+    return params.query(
+        'filters',
+        schema.optional(application_json(filter_schema)),
+        description=format_description('''
+            Criteria to filter entities from the search results.
+
+            Each filter consists of a facet name, a relational operator, and an
+            array of facet values. The available operators are "is", "within",
+            "contains", and "intersects". See the `sort` parameter for a list of
+            filter-able facets.
+
+            For the "is" operator, multiple facet values are combined using "or"
+            logic. For example, `{"fileFormat": {"is": ["fastq", "fastq.gz"]}}`
+            selects entities where the file format is either "fastq" or
+            "fastq.gz". For the other operators, the facet values must come
```

Comment:
```suggestion
            "fastq.gz". For the "within", "intersects" and "contains" operators, the facet values must come
```

We may add other operators.